src/foam/lib/csv/CSVOutputter.js

```diff
@@ -0,0 +1,227 @@
+        this.isFirstColumn = false;
+        this.outputValue_(value);
+      },
+      javaCode: `
+        if ( ! getIsFirstColumn() ) getSb().append(",");
+        setIsFirstColumn(false);
+        outputValue_(value);
+      `
+    },
+    {
+      name: 'outputValue_',
+      args: [
+        { type: 'Any', name: 'value' }
+      ],
+      code:
+        foam.mmethod(
+          {
+            String: function(value) {
+              this.csv += `"${value.replace(/\"/g, '""')}"`;
+            },
+            Number: function(value) {
+              this.csv += value.toString();
+            },
+            Boolean: function(value) {
+              this.csv += value.toString();
+            },
+            Date: function(value) {
+              this.outputValue_(value.toDateString());
+            },
+            FObject: function(value) {
+              this.outputValue_(foam.json.Pretty.stringify(value));
+            },
+            Array: function(value) {
+              this.outputValue_(foam.json.Pretty.stringify(value));
+            },
+            Undefined: function(value) {},
+            Null: function(value) {}
+          }, function(value) {
+            this.outputValue_(value.toString());
+        }),
+      javaCode: `
+        if ( value instanceof String ) {
+          getSb().append("\\"");
+          getSb().append(((String)value).replace("\\"", "\\"\\""));
+          getSb().append("\\"");
+        } else if ( value instanceof Number ) {
+          getSb().append(value.toString());
+        } else if ( value instanceof Boolean ) {
+          getSb().append(value.toString());
+        } else if ( value instanceof Date ) {
+          getSb().append(value.toString());
+        } else if ( value instanceof FObject ) {
+          outputValue_(value.toString());
+        } else if ( value instanceof List ) {
+          outputValue_(value.toString());
+        } else {
+          outputValue_(value.toString());
+        }
+      `
+    },
+    {
+      name: 'newLine_',
+      code: function() {
+        this.csv += '\n';
+        this.isFirstColumn = true;
+      },
+      javaCode: `
+        getSb().append("\\n");
+        setIsFirstColumn(true);
+      `
+    },
+    {
+      name: 'toString',
+      code: function() {
+        return this.csv;
+      },
+      javaCode: 'return getSb().toString();'
+    },
+    {
+      name: 'headerOutput',
+      args: [
+        { type: 'FObject', name: 'obj' }
+      ],
+      code: function() {
+        var element;
+        this.props.forEach((name) => {
+          element = this.of.getAxiomByName(name);
+          element.toCSVLabel(this, element);
+        });
+        this.newLine_();
+        this.isFirstRow = false;
+      },
+      javaCode: `
+        Object propObj;
+        PropertyInfo columnProp;
+        String[] tableColumnNames = getProps();
+
+        for (String propName: tableColumnNames) {
+          propObj = obj.getProperty(propName);
```

Comment:
Since this code block is basically identical to the JS version, try to use the same variable names for the same object if you can. columnProp would be renamed to axiom or prop depending on what you renamed it to above.
propObj shouldn't be necessary.